src/test/suite/IntegrationUtils.test.ts

```diff
@@ -19,48 +19,69 @@
 
-    let sandbox: sinon.SinonSandbox;
-    let showQuickPickStub: sinon.SinonStub;
-    let executeTaskStub: sinon.SinonStub;
-    
-    setup(() => {
+	let sandbox: sinon.SinonSandbox;
+	let showQuickPickStub: sinon.SinonStub;
+	let executeTaskStub: sinon.SinonStub;
+	let newIntegrationStub: sinon.SinonStub;
+
+	setup(() => {
 		sandbox = sinon.createSandbox();
-        showQuickPickStub = sandbox.stub(vscode.window, 'showQuickPick');
-        executeTaskStub = sandbox.stub(vscode.tasks, 'executeTask');
-	});	
+		showQuickPickStub = sandbox.stub(vscode.window, 'showQuickPick');
+		executeTaskStub = sandbox.stub(vscode.tasks, 'executeTask');
+		newIntegrationStub = sandbox.stub(IntegrationUtils, 'createNewIntegration');
+	});
 
 	teardown(() => {
-        showQuickPickStub.restore();
-        executeTaskStub.restore();
+		showQuickPickStub.restore();
+		executeTaskStub.restore();
+		newIntegrationStub.restore();
 		sandbox.reset();
 	});
 
-    test("ensure listing Camel K task when accessing 'Start Apache Camel Integration'", async function() {
-        showQuickPickStub.onFirstCall().returns(IntegrationUtils.vscodeTasksIntegration);
-        showQuickPickStub.onSecondCall().returns('Test Camel K task');
-        
-        await IntegrationUtils.startIntegration(getDocUri('MyRouteBuilder.java'));
-        
-        sinon.assert.calledWith(showQuickPickStub,
-            ['Start in dev mode Camel K integration opened in active editor', 'Test Camel K task'],
-            {placeHolder: 'Choose a predefined task'});
-        sinon.assert.calledOnce(executeTaskStub);
-    });
+	test("ensure listing Camel K task when accessing 'Start Apache Camel Integration'", async function () {
+		showQuickPickStub.onFirstCall().returns(IntegrationUtils.vscodeTasksIntegration);
+		showQuickPickStub.onSecondCall().returns('Test Camel K task');
+
+		await IntegrationUtils.startIntegration(getDocUri('MyRouteBuilder.java'));
+
+		sinon.assert.calledWith(showQuickPickStub,
+			['Start in dev mode Camel K integration opened in active editor', 'Test Camel K task'],
+			{ placeHolder: 'Choose a predefined task' });
+		sinon.assert.calledOnce(executeTaskStub);
+	});
+
+	test("ensure filtering out Camel K task with unrelated target file when accessing 'Start Apache Camel Integration'", async function () {
+		showQuickPickStub.onFirstCall().returns(IntegrationUtils.vscodeTasksIntegration);
+		showQuickPickStub.onSecondCall().returns(undefined);
 
-    test("ensure filtering out Camel K task with unrelated target file when accessing 'Start Apache Camel Integration'", async function() {
-        showQuickPickStub.onFirstCall().returns(IntegrationUtils.vscodeTasksIntegration);
-        showQuickPickStub.onSecondCall().returns(undefined);
-        
-        await IntegrationUtils.startIntegration(getDocUri('ADifferentRouteBuilder.java'));
+		try {
+			await IntegrationUtils.startIntegration(getDocUri('ADifferentRouteBuilder.java'));
+		} catch (onrejected) {
```

Comment:
I am confused by the naming of the error variable. It pretends to be some callback function but its just a plain error var. How about renaming it to something more meaningful?

Reply:
I'm using the naming of the documentation. Seems to be the most coherent. What would be more meaningful?

Reply:
What documentation are you referring to?

Reply:
argh, it was the then().catch() documentation. Good *catch* :-D